lib/lower/convert-type.cc

```diff
@@ -488,3 +488,11 @@ M::FunctionType Br::translateSymbolToFIRFunctionType(
 M::Type Br::convertReal(M::MLIRContext *context, int kind) {
   return genFIRType<RealCat>(context, kind);
 }
+
+M::Type Br::getSequenceRefType(M::Type refType) {
+  auto type{refType.dyn_cast<fir::ReferenceType>()};
+  assert(type && "expected a reference type");
+  auto elementType{type.getEleTy()};
+  fir::SequenceType::Shape shape{-1};
```

Comment:
What does a hardcoded -1 mean here?

Reply:
It is the way I managed to get an unknown extent, what is the cleaner way ?

Reply:
Should I define a `static constexpr Extent unkownExtent = -1` member in SequenceType class https://github.com/schweitzpgi/f18/blob/f18/include/fir/FIRType.h#L249 or is it already defined somewhere else ?

Reply:
An unknown Shape (i.e., <* x i32>) should just be encoded as an empty vector. The -1 is correct if the shape is known but the extent isn't (i.e., <? x i32>).

Reply:
But I like your suggestion better than hard-coded -1 constants in the sources.